backend/layers/business/business.py

```diff
@@ -576,11 +578,28 @@ def ingest_dataset(
             }
         )
 
-        # TODO: validate all uris in the manifest
-        # TODO: replace the uris with the actual uri if a uri to an existing h5ad or fragments file is provided
-        for _url in urls:
+        # Validate the URIs
+        # TODO: This should be done in the IngestionManifest class
+        for key, _url in manifest.model_dump(exclude_unset=True).items():
             if not self.uri_provider.validate(_url):
                 raise InvalidURIException(f"Trying to upload invalid URI: {_url}")
+            if not self.is_public_uri(_url):
+                continue
+            if not current_dataset_version_id:
+                raise InvalidIngestionManifestException(
+                    "Cannot ingest public datasets without a current dataset version"
+                )
+            if key == "anndata":
+                dataset_version_id, extension = url.split("/")[-1].split(".")
```

Comment:
This will error (and throw a 500) if a fragment file has been passed (since `.split(".")` would return 3 items).

```suggestion
                dataset_version_id, extension = url.split("/")[-1].split(".", maxsplit=1)
```

This should allow something like `.tsv.gz` for extension